pex/pex.py

```diff
@@ -94,6 +94,18 @@ def _extras_paths(cls):
     extras_paths = filter(None, makefile.get('EXTRASPATH', '').split(':'))
     for path in extras_paths:
       yield os.path.join(standard_lib, path)
+    # macOS Sierra support
+    framework = sysconfig.get_config_var("PYTHONFRAMEWORK")
+    if not framework:
+      raise StopIteration()
+    third_party = os.path.join("/Library", framework, sys.version[:3], "site-packages")
```

Comment:
this seems like a poor way of determining the `site-packages` dir - it hardcodes `/Library` and extracts version information from the `sys.version` string (rather than say, `sys.version_info`).

instead, prefer iterating over entries provided by the `site` module's `getsitepackage()`:

```
>>> import site; site.getsitepackages()
['/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages', '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/site-python', '/Library/Python/2.7/site-packages']
```

which already has an existing helper class method in `PEX._get_site_packages()`.

this probably also obviates the need to fetch the 'PYTHONFRAMEWORK' config var?